engine/primitive/src/main/java/io/deephaven/engine/primitive/iterator/CloseableIterator.java

```diff
@@ -17,14 +16,14 @@
 import java.util.stream.StreamSupport;
 
 /**
- * This interface extends {@link Iterator} and {@link SafeCloseable} in order to allow for iterators that acquire
+ * This interface extends {@link Iterator} and {@link AutoCloseable} in order to allow for iterators that acquire
  * resources that must be released. Such iterators must document this need, and appropriate measures (e.g. a
  * try-with-resources block) should be taken to ensure that they are {@link #close() closed}. Methods that return
  * streams over CloseableIterator instances should ensure that closing the resulting {@link Stream},
  * {@link java.util.stream.IntStream IntStream}, {@link java.util.stream.LongStream LongStream}, or
  * {@link java.util.stream.DoubleStream DoubleStream} will also close the iterator.
  */
-public interface CloseableIterator<TYPE> extends Iterator<TYPE>, SafeCloseable {
```

Comment:
Do we want stricter guarantee of idempotentcy w/ `Closeable`? Not a blocker. Happy with any answer: yes, no, or yes, but later.